general/management/commands/prune_database.py

```diff
@@ -95,23 +110,36 @@ def delete_some_users(self, userids):
             cursor.execute("delete from sounds_packdownloadsound where pack_download_id in (select id from sounds_packdownload where user_id in %s)", [tuple(userids)])
             cursor.execute("delete from sounds_packdownload where user_id in %s", [tuple(userids)])
         console_logger.info('   - done, user objects')
-        # This will delete some other related data, but it's not as slow as deleting downloads.
-        # so we let django do it
         User.objects.filter(id__in=userids).delete()
         console_logger.info('   - done')
 
-    def delete_sound_uploaders(self, pkeep):
-        """Delete some percentage of users who have uploaded sounds
+    def delete_sound_uploaders(self, numkeep):
+        """Delete users who have uploaded sounds, keeping at least `numkeep` sounds.
+           Don't consider users who have uploaded more than 1000 sounds.
            Arguments:
-              pkeep: the percentage of uploaders to keep
+              numkeep: the number of sounds to keep
         """
         console_logger.info('Deleting some uploaders')
-        userids = User.objects.values_list('id', flat=True).filter(profile__num_sounds__gt=0)
-        numusers = len(userids)
+        users = User.objects.values_list('id', 'profile__num_sounds').filter(profile__num_sounds__gt=0)
+        users = list(users)
+        all_users_with_sounds = [u[0] for u in users]
+        random.shuffle(users)
+        numusers = len(users)
         console_logger.info(f'Number of uploaders: {numusers}')
-        percentage_remove = 1.0 - (pkeep / 100.0)
-        randusers = sorted(random.sample(userids, int(numusers*percentage_remove)))
-        ch = [c for c in chunks(randusers, 100)]
+
+        totalsounds = 0
+        userids = []
+        for u, numsounds in users:
+            if numsounds <= 1000:
```

Comment:
should be a config option

Reply:
after looking at this again, it doesn't seem important to have this an argument. I've moved it to an in-file constant